machine.go

```diff
@@ -707,6 +708,23 @@ func (m *Machine) UpdateGuestDrive(ctx context.Context, driveID, pathOnHost stri
 	return nil
 }
 
+// IsAlive makes sure that a Firecracker instance is running and responds via API by
+// calling `DescribeInstance` endpoint behind.
+func (m *Machine) IsAlive(ctx context.Context) error {
+	instance, err := m.client.DescribeInstance(ctx)
+	if err != nil {
+		return err
+	}
+
+	state := StringValue(instance.Payload.State)
```

Comment:
I don't think it makes sense to verify the state if the purpose of this method is to see whether Firecracker responds to an API.  I think getting any non-error response would meet the success criteria.

Reply:
If the motivation behind the change is to test the responsiveness of the API, then agreed and also think `IsVMMResponsive` is a good name.

If the motivation is instead to test that the Machine is in the running state, then can we just add a `State` function that returns the state value, whatever it may be?

Reply:
@haikuoliu which behavior would suite better here?

Reply:
Right now I've renamed it to `IsRunning` which tells whether a VM is in `Starting` or `Running` state.

Reply:
It is a bit confusing to see that we have `Running` as a state, and `IsRunning` actually means `IsRunningOrStarting`. Naming is hard...

I'm inclined to expose Firecracker's state as is instead of introducing our own notion of the state (running, responsive, alive, ...).

Reply:
The status type could provide rich methods later, like https://godoc.org/inet.af/http#Status.IsClientError (inet.af is bradfitz's net/http rework proposal, which is referenced from https://github.com/golang/go/issues/29652)